apps/server/src/modules/files-storage/uc/files-storage-delete.uc.spec.ts

```diff
@@ -0,0 +1,333 @@
+
+				return { requestParams, userId, error };
+			};
+
+			it('should throw error if entity not found', async () => {
+				const { userId, requestParams, error } = setup();
+
+				await expect(filesStorageUC.deleteOneFile(userId, requestParams)).rejects.toThrow(error);
+			});
+		});
+
+		describe('WHEN user is authorized', () => {
+			const setup = () => {
+				const { fileRecords, userId } = getFileRecordsWithParams();
+				const fileRecord = fileRecords[0];
+				const requestParams = { fileRecordId: fileRecord.id, parentType: fileRecord.parentType };
+
+				filesStorageService.getFileRecord.mockResolvedValueOnce(fileRecord);
+
+				return { requestParams, userId, fileRecord };
+			};
+
+			it('should call authorizationService.checkPermissionByReferences', async () => {
+				const { requestParams, userId, fileRecord } = setup();
+
+				await filesStorageUC.deleteOneFile(userId, requestParams);
+
+				const allowedType = FilesStorageMapper.mapToAllowedAuthorizationEntityType(requestParams.parentType);
+
+				expect(authorizationService.checkPermissionByReferences).toBeCalledWith(
+					userId,
+					allowedType,
+					fileRecord.parentId,
+					PermissionContexts.delete
+				);
+			});
+		});
+
+		describe('WHEN file is found', () => {
+			const setup = () => {
+				const { fileRecords, userId } = getFileRecordsWithParams();
+				const fileRecord = fileRecords[0];
+				const requestParams = { fileRecordId: fileRecord.id };
+
+				filesStorageService.getFileRecord.mockResolvedValueOnce(fileRecord);
+
+				return { requestParams, userId };
+			};
+
+			it('should call getFile once', async () => {
+				const { userId, requestParams } = setup();
+
+				await filesStorageUC.deleteOneFile(userId, requestParams);
+
+				expect(filesStorageService.getFileRecord).toHaveBeenCalledTimes(1);
+			});
+
+			it('should call getFile with correctly params', async () => {
+				const { userId, requestParams } = setup();
+
+				await filesStorageUC.deleteOneFile(userId, requestParams);
+
+				expect(filesStorageService.getFileRecord).toHaveBeenCalledWith(requestParams);
+			});
+		});
+
+		describe('WHEN user is not authorized', () => {
+			const setup = () => {
+				const { fileRecords, userId } = getFileRecordsWithParams();
+				const fileRecord = fileRecords[0];
+				const requestParams = { fileRecordId: fileRecord.id, parentType: fileRecord.parentType };
+
+				filesStorageService.getFileRecord.mockResolvedValueOnce(fileRecord);
+				authorizationService.checkPermissionByReferences.mockRejectedValueOnce(new ForbiddenException());
+
+				return { requestParams, userId };
+			};
+
+			it('should throw forbidden exception', async () => {
+				const { requestParams, userId } = setup();
+
+				await expect(filesStorageUC.deleteOneFile(userId, requestParams)).rejects.toThrow(new ForbiddenException());
+				expect(filesStorageService.delete).toHaveBeenCalledTimes(0);
+			});
+		});
+
+		describe('WHEN delete was successful', () => {
```

Comment:
'WHEN delete was successful' and 'WHEN user is authorized' should be the same setup and i am very confusing why no mock for authorisation service is needed in both. Maybe the deep mock on this place bring us in trouble. 
For the server application authrisation is syncron and need no DB call and no mock for valid cases, but in our case it is async because the modul load the ressources and we need the mock.